### PR TITLE
[IOTDB-8] Fix filenode restore deserialization

### PR DESCRIPTION
**Pull request by liukun4515**
https://issues.apache.org/jira/browse/IOTDB-8

### CODE REVIEW

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessorStore.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessorStore.java]: bad indent

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/OverflowChangeType.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *     http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/OverflowChangeType.java]: bad indent

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/TsFileResource.java [hunk 3]
```diff
@@ -7,7 +7,7 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
- *      http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/TsFileResource.java]: bad indent

File: iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/TsFileResource.java [hunk 4]
```diff
@@ -96,6 +97,64 @@ public TsFileResource(OverflowChangeType type, String relativePath) {
     this(type, 0, relativePath);
   }
 
+  public void serialize(OutputStream outputStream) throws IOException {
+    ReadWriteIOUtils.write(this.overflowChangeType.serialize(), outputStream);
+    ReadWriteIOUtils.write(this.baseDirIndex, outputStream);
+    if (this.relativePath != null) {
+      ReadWriteIOUtils.writeIsNull(this.relativePath, outputStream);
+      ReadWriteIOUtils.write(this.relativePath, outputStream);
+    }
```
Comment (jixuan1989) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/TsFileResource.java]: Hm.... I guess the correct codes are:
```
    ReadWriteIOUtils.writeIsNull(this.relativePath, outputStream);
    if (this.relativePath != null) {
      ReadWriteIOUtils.write(this.relativePath, outputStream);
    }
```

File: iotdb/src/test/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessorStoreTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jixuan1989) [inline iotdb/src/test/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessorStoreTest.java]: bad indent

File: iotdb/src/test/java/org/apache/iotdb/db/engine/filenode/TsFileResourceTest.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
```
Comment (jixuan1989) [inline iotdb/src/test/java/org/apache/iotdb/db/engine/filenode/TsFileResourceTest.java]: bad indent

File: iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowResource.java [hunk 7]
```diff
@@ -127,7 +127,7 @@ public OverflowResource(String parentPath, String dataPath, VersionController ve
       if (insertTempFile.exists()) {
         left = insertTempFile.length();
       }
-      LOGGER.warn("Cannot read position info, returning a default value", e);
+      LOGGER.warn("Cannot read position info, because of {}, returning a default value", e.getMessage());
```
Comment (jt2594838) [inline iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowResource.java]: You should move the reason to the end and remove the bracelets.

**Pull request merged**